pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ecdsasecp256k1signature2019
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/btcsuite/btcd/btcec"
+
+	sigverifier "github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+)
+
+const (
+	secp256k1KeySize = 32
+)
+
+// PublicKeyVerifier verifies a secp256k1 signature taking public key bytes and JSON Web Key as input.
+// NOTE: this verifier is present for backward compatibility reasons and can be removed later.
+// Please use CryptoVerifier or your own verifier.
+type PublicKeyVerifier struct {
+}
+
+// Verify will verify a signature.
+func (v *PublicKeyVerifier) Verify(pubKey *sigverifier.PublicKey, msg, signature []byte) error {
+	err := v.validatePublicKey(pubKey)
+	if err != nil {
+		return err
+	}
+
+	if len(signature) != 2*secp256k1KeySize {
+		return errors.New("ecdsa: invalid signature size")
+	}
+
+	curve := btcec.S256()
+
+	btcecPubKey, err := btcec.ParsePubKey(pubKey.Value, curve)
+	if err != nil {
+		return errors.New("ecdsa: invalid public key")
+	}
+
+	ecdsaPubKey := btcecPubKey.ToECDSA()
+	hasher := crypto.SHA256.New()
+
+	_, err = hasher.Write(msg)
+	if err != nil {
+		return errors.New("ecdsa: hash error")
+	}
+
+	hash := hasher.Sum(nil)
+
+	r := big.NewInt(0).SetBytes(signature[:secp256k1KeySize])
+	s := big.NewInt(0).SetBytes(signature[secp256k1KeySize:])
+
+	verified := ecdsa.Verify(ecdsaPubKey, hash, r, s)
+	if !verified {
+		return errors.New("ecdsa: invalid signature")
+	}
+
+	return nil
+}
+
+func (v *PublicKeyVerifier) validatePublicKey(pubKey *sigverifier.PublicKey) error {
+	// A presence of JSON Web Key is mandatory (due to EcdsaSecp256k1VerificationKey2019 type).
+	if pubKey.JWK == nil {
+		return ErrJWKNotPresent
+	}
+
+	if pubKey.Type != jwkType {
+		return ErrTypeNotEcdsaSecp256k1VerificationKey2019
+	}
+
+	if pubKey.JWK.Kty != "EC" {
+		return fmt.Errorf("unsupported key type: '%s'", pubKey.JWK.Kty)
+	}
+
+	if pubKey.JWK.Crv != "" && pubKey.JWK.Crv != "secp256k1" {
```

Comment:
`pubKey.JWK.Crv != ""` is not needed, Crv must be `secp256k1` regardless

Reply:
`crv` and `alg` are not mandatory fields of JSON Web Key (https://tools.ietf.org/html/rfc7517). That's why I put a soft check here.  For example,  at [reference impl](https://github.com/decentralized-identity/lds-ecdsa-secp256k1-2019.js) they use `kty` and `crv` only: https://github.com/decentralized-identity/lds-ecdsa-secp256k1-2019.js/blob/461bc062107b262e15c945b4fd02ca2e1ab22c27/packages/lds-ecdsa-secp256k1-2019-demo/src/__tests__/did-elem.spec.js#L18

Reply:
ok

Reply:
OK, I've re-checked it, `crv` is mandatory for Elliptic Curve Keys, I will improve that: https://tools.ietf.org/html/draft-ietf-jose-json-web-key-00#section-4.2.1

Reply:
Fixed, thank you @Baha-sk for the review comment!

Reply:
I think Algorithm is also required to specify the bitcoin alg below, you can remove the `Algorithm != ""` condition from there too to be consistent with the other checks in this function.

Reply:
I conceptually agree with you. But we want to be compatible with other implementations, e.g. https://github.com/decentralized-identity/lds-ecdsa-secp256k1-2019.js/blob/461bc062107b262e15c945b4fd02ca2e1ab22c27/packages/lds-ecdsa-secp256k1-2019-demo/src/__tests__/did-elem.spec.js#L18. Also, `alg` is not mandatory at https://tools.ietf.org/html/rfc7517. So let's relax it.